default/python/AI/ProductionAI.py

```diff
@@ -1402,10 +1402,37 @@ def generate_production_orders():
                 if (not prioritized) and (this_priority == PriorityType.PRODUCTION_INVASION):
                     fo.issueRequeueProductionOrder(production_queue.size - 1, 0)  # move to front
         print
+    update_stockpile_use()
     fo.updateProductionQueue()
     _print_production_queue(after_turn=True)
 
 
+def update_stockpile_use():
+    """Decide which elements in the production_queue will be enabled for drawing from the imperial stockpile.  This
+    initial version simply ensures that every resource group with at least one item on the queue has its highest
+    priority item be stockpile-enabled.
+
+    :return: None
+    """
+    # TODO: Do a priority and risk evaluation to decide on enabling stockpile draws
+    empire = fo.getEmpire()
+    production_queue = empire.productionQueue
+    resource_groups = set(tuple(el.key()) for el in empire.planetsWithAvailablePP)
+    planets_with_group_stockpile_elem = set()
+    for queue_index in range(len(production_queue)):
+        element = production_queue[queue_index]
+        if element.locationID in planets_with_group_stockpile_elem:
+            continue
+        group = None
+        for group in resource_groups:
+            if element.locationID in group:
+                break
+        else:
+            continue # we don't appear to own the location any more
```

Comment:
Use built-in ```next```:
```
group = next(x for x in resource_groups if element.locationID in x, None)
if group is None:
    continue
```

Reply:
with some extra parentheses that will work, ok